nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/test/java/org/apache/nifi/marklogic/processor/QueryMarkLogicIT.java

```diff
@@ -96,10 +126,169 @@ public void testSimpleCollectionQuery() throws InitializationException {
         assertTrue(Arrays.equals(expectedByteArray, actualByteArray));
     }
 
+    @Test
+    public void testCombinedJSONQuery() throws InitializationException {
+        TestRunner runner = getNewTestRunner(QueryMarkLogic.class);
+        runner.setProperty(QueryMarkLogic.QUERY, "{\"search\" : {\n" +
+                "  \"ctsquery\": {\n" +
+                "    \"jsonPropertyValueQuery\":{\n" +
+                "      \"property\":[\"sample\"],\n" +
+                "      \"value\":[\"jsoncontent\"]\n" +
+                "      } \n" +
+                "  }\n" +
+                "} }");
+        runner.setProperty(QueryMarkLogic.QUERY_TYPE, QueryMarkLogic.QueryTypes.COMBINED_JSON.name());
+        runner.assertValid();
+        runner.run();
+        // mod 3 == 0 docs are JSON, but mod 5 == 0 docs are XML and take precedence in doc generation
+        int expectedSize = (int) (Math.ceil((numDocs - 1.0) / 3.0) - Math.ceil((numDocs - 1.0) / 15.0));
```

Comment:
I think it would be better to have this logic in a method on the parent class, in the same place as the logic that makes it work (if that ever changes, this needs to also). Something like:

int expectedSize = getDocumentCount("json");